File: algorithms/1_easy/T/00867_transpose_matrix/solution.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        const size_t rows{matrix.size()};
        const size_t cols{matrix[0].size()};
        vector<vector<int>> res(cols, vector<int>(rows, 0));

        for (size_t j{0}; j < cols; ++j) {
            for (size_t i{0}; i < rows; ++i) {
                res[j][i] = matrix[i][j];
            }
        }

        return res;
    }
};

File: algorithms/1_easy/T/00867_transpose_matrix/solution.rs
impl Solution {
    pub fn transpose(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let rows = matrix.len();
        let cols = matrix[0].len();
        let mut res = vec![vec![0_i32; rows]; cols];

        for j in 0..cols {
            for i in 0..rows {
                res[j][i] = matrix[i][j];
            }
        }

        res
    }
}
